build/script.js

```diff
@@ -102,6 +104,34 @@ gulp.task('scripts-watch', createScriptsStream(false));
  * 'async'.
  */
 function createCompileTask(translation) {
+
+  let outputDir =
+      translation ? path.join(conf.paths.i18nProd, `/${translation.key}`) : conf.paths.prodTmp;
+
+  return (
+      (next) =>
+          gulp.src([
+                // Application source files.
+                path.join(conf.paths.frontendSrc, '**/*.js'),
+                // Partials generated by other tasks, e.g., Angular templates.
+                path.join(conf.paths.partials, '**/*.js'),
+                // Include base.js to enable some compiler functions, e.g., @export annotation
+                // handling and getMsg() translations.
+                path.join(
+                    conf.paths.bowerComponents, 'google-closure-library/closure/goog/base.js'),
+              ])
+              .pipe(closureCompiler(translation))
+              .pipe(gulpIf('**/app.js',gulpReplace('BUILD_DASHBOARD_VERSION', conf.deploy.version.release)))
+              .pipe(gulpIf('**/app.js',gulpReplace('BUILD_GIT_COMMIT', process.env.TRAVIS_COMMIT)))
```

Comment:
Changing `app.js` (whole aplication source) is very risky and should be avoided if it's possible. We should search for other solutions, doing it before build seems to be better idea, but not the only one. Moreover, if it will be done before minimizing to `app.js` it cannot change sources in `/src` directory.